subsys/bluetooth/audio/bap_scan_delegator.c

```diff
@@ -225,33 +262,46 @@ static void net_buf_put_recv_state(const struct bass_recv_state_internal *recv_s
 }
 
 static void receive_state_updated(struct bt_conn *conn,
-				  const struct bass_recv_state_internal *internal_state)
+				  struct bass_recv_state_internal *internal_state)
 {
+#if CONFIG_BT_BAP_SCAN_DELEGATOR_LOG_LEVEL >= LOG_LEVEL_DBG
 	int err;
 
-	/* If something is holding the NOTIFY_PEND flag we should not notify now */
-	if (atomic_test_bit(internal_state->flags,
-			    BASS_RECV_STATE_INTERNAL_FLAG_NOTIFY_PEND)) {
-		return;
+	err = k_mutex_lock(&internal_state->mutex, SCAN_DELEGATOR_BUF_SEM_TIMEOUT);
+	if (err == 0) {
+		bt_debug_dump_recv_state(internal_state);
+		err = k_mutex_unlock(&internal_state->mutex);
+		__ASSERT(err == 0, "Failed to unlock mutex: %d", err);
+
+	} else {
+		LOG_DBG("Failed to lock mutex");
```

Comment:
General comment: Suggest to do this for all of these lines:
```suggestion
		LOG_DBG("Failed to lock mutex: %d", err);
```